src/unit-test-coverage/shared/src/coveragetest-idmap.c

```diff
@@ -1065,6 +1070,76 @@ void Test_OS_ObjectIdIterator(void)
     UtAssert_STUB_COUNT(OS_Unlock_Global_Impl, 2);
 }
 
+void Test_OS_ObjectIDInteger(void)
+{
+    /*
+     * Test Case For:
+     * OS_ObjectIdToInteger, OS_ObjectIdFromInteger, OS_ObjectIdEqual, OS_ObjectIdDefined
+     */
+    int32             actual;
+    OS_object_token_t token;
+    osal_id_t         typesI[maxRecords];
+    osal_id_t         typesJ[maxRecords];
+    uint32            intID;
+    int32             recordscount = 0;
+    uint32            maxType      = 0;
+    uint32            typeReturn   = 0;
+    osal_objtype_t    idtype;
+    char              str[maxRecords];
+
+    for (idtype = 0; idtype < OS_OBJECT_TYPE_USER; ++idtype)
+    {
+        typeReturn = OS_GetMaxForObjectType(idtype);
+        maxType += typeReturn;
+    }
```

Comment:
What is the philosophy here with adding up all the max values?  Why not check each one individually?  For instance if there was a bug that caused two values to get swapped, this would not catch it, because it would still add up to the same result.

Reply:
Part of it is to confirm the total calculation in the test matches the calculation in code, since there isn't a total available externally, and these should never get out of sync.  Alternative would be to expose the total.